File: Refrigtz/FormSelect.cs
using System;
using System.IO;
using System.Windows.Forms;
namespace Refrigtz
{
    [Serializable]
    public partial class FormSelect : Form
    {
        public static int OrderPlateDraw = 1;
        static void Log(Exception ex)
        {
            try
            {
                Object a = new Object();
                lock (a)
                {
                    string stackTrace = ex.ToString();
                    File.AppendAllText(FormRefrigtz.Root + "\\ErrorProgramRun.txt", stackTrace + ": On" + DateTime.Now.ToString()); // path of file where stack trace will be stored.}
                }
            }
            catch (Exception t) { Log(t); }
        }
        //Constructor.
        public FormSelect()
        {
            InitializeComponent();
        }
        //Delegate Of Form Close Visibility.
        delegate void SetCloseVisibleCallback();

        public void SetCloseVisible()
        {
            Object O = new Object();
            lock (O)
            {         // InvokeRequired required compares the thread ID of the
                      // calling thread to the thread ID of the creating thread.
                      // If these threads are different, it continue;s true.
                if (this.InvokeRequired)
                {
                    try
                    {

                        SetCloseVisibleCallback d = new SetCloseVisibleCallback(SetCloseVisible);
                        this.Invoke(new Action(() => this.Close()));
                    }
                    catch (Exception t) { Log(t); }
                }
                else
                {
                    try
                    {
                        this.Close();
                    }
                    catch (Exception t) { Log(t); }
                }
            }
        }
        private void FormSelect_Load(object sender, EventArgs e)
        {
            Object O = new Object();
            lock (O)
            {
                if (File.Exists("_DonotDelete.txt"))
                    RadioButtonBrownOrder.Checked = true;
                else
                    RadioButtonGrayOrder.Checked = true;


            }
        }

        private void RadioButtonBrownOrder_CheckedChanged(object sender, EventArgs e)
        {
            Object O = new Object();
            lock (O)
            {
                OrderPlateDraw = -1;
                if (RadioButtonBrownOrder.Checked)
                {
                    if (!File.Exists("_DonotDelete.txt"))
                        File.Create("_DonotDelete.txt");

                }
            }
        }

        private void RadioButtonGrayOrder_CheckedChanged(object sender, EventArgs e)
        {
            Object O = new Object();
            lock (O)
            {
                if (RadioButtonGrayOrder.Checked)
                {
                    OrderPlateDraw = 1;
                    if (File.Exists("_DonotDelete.txt"))
                        File.Delete("_DonotDelete.txt");
                }
            }
        }
    }
}
//End Documents.

File: Refrigtz/FormSelect.vb
Public Class FormSelect


    Public Shared Sub Log(ex As Exception)

        Try

            Dim StackTrace As String = ex.ToString()
            IO.File.AppendAllText("ErrorProgramRun.txt", StackTrace + ": On" + DateTime.Now.ToString())
            ' path Of file where stack trace will be stored.

        Catch t As Exception
            Log(t)
        End Try
    End Sub
    'Constructor.
    'Delegate Of Form Close Visibility.
    Delegate Sub SetCloseVisibleCallback()
    Private Sub SetClose()

        ' InvokeRequired required compares the thread ID of the
        ' calling thread to the thread ID of the creating thread.
        ' If these threads are different, it returns true.
        If Me.IsAccessible Then
            Dim d As New SetCloseVisibleCallback(AddressOf SetClose)
            Me.Close()
        Else
            Me.Close()
        End If
    End Sub

    Private Sub FormSelect_Load(sender As Object, e As EventArgs) Handles MyBase.Load

    End Sub
End Class